mass_mailing_partner/models/mail_mass_mailing_contact.py

```diff
@@ -75,7 +82,9 @@ def _set_name_email(self, vals):
         partner_id = vals.get('partner_id', self.partner_id.id)
         if not partner_id:
             return vals
-        partner = self.env['res.partner'].browse(partner_id)
-        vals['email'] = partner.email
-        vals['name'] = partner.name
+        partner = self.env['res.partner'].browse(partner_id, self._prefetch)
```

Comment:
Why do you make the call positional here and named here https://github.com/OCA/social/pull/315/files#diff-6e2abeeb2e2cc273679a065d55fcd860R67 ?

Reply:
I used positional at first, but they asked me to change it above, so I changed it. Should I change it here also?

Reply:
I don't know :man_shrugging:

Reply:
Keyword-only arguments are a python3-only feature that is not used in Odoo AFAIK.

The signature for keyword-only arguments is:

```python
def method(self, method, whatever=True, *, keyword_only=False):
   pass
```

By "keyword" did you mean that they have a default value?

Reply:
Yes, arguments with a default value are called "keyword arguments". `keyword_only` arguments are for explicitly forcing them to be used with keyword, but a good practice is to always use keyword arguments with the argument, so that if someone use another order (or arguments are passed as dictionaries with `**kwargs`), they are kept in the proper dictionary or argument.

Reply:
Usually I use keywords when the argument is not explicit enough. In this case, `self._prefetch` is quite obviously passed to the prefetch argument.

Nevermind, it's changed.

Reply:
But do you understand the reasoning about keeping them keywords? Imagine you have an overriding like that:
```python
def browse(args, kwargs):
    if 'prefetch' in kwargs:
        # do stuff
    return super().browse(*args, **kwargs)
```
it won't do what you want, and you have to hardcode possible argument position in `args`.

Reply:
It's not a bug, but a generic way to catch any argument. What you write implies that you have to add all the positional arguments previous to the keyword argument you pass as positional and also that you add keyword arguments in the same order, which is not true before P3.6 when you pass a dictionary to super.

Reply:
Well, you're right, it's not a bug. However, your code would only work if the `prefetch` argument is passed as a keyword one. If that's what you really want, then that's the way to do it. If what you really want is to get the parameter, no matter how the method is called, then you need to write the paramteres in order as I did. That's normal python (in fact, even python 2), and has nothing to do with py3.6 dict order AFAIK.

The fact that an argument has a `=Whatever` in its signature means that the argument has a default value, _not_ that you must pass it as a keyword argument. Otherwise, Python wouldn't have a way to pass default values to ordered arguments.

In any case, it's changed as you wanted, so I guess this whole conversation is unnecessary.